src/components/SurveyField/SurveyField.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import styled from 'styled-components';
+import { arrayOf, string, func } from 'prop-types';
+import { Grommet } from 'grommet';
+
+import { spacingLarge, colorRed, colorBlue } from '../../styles/designTokens';
+import { formField, answerType } from '../../types';
+import RatingField from './RatingField';
+import ShortInputField from './ShortInputField';
+import LongInputField from './LongInputField';
+import SelectField from './SelectField';
+import YesNoField from './YesNoField';
+import media from '../../styles/media';
+
+const Wrapper = styled.div`
+  margin: 3rem 0;
+  position: relative;
+
+  span[data-type='required-field'] {
+    position: absolute;
+    top: 0.3rem;
+    color: ${colorRed};
+  }
+
+  ${media.mobileM`
+    margin: 4.5rem 0rem;
+  `}
+
+  ${media.mobileL`
+    margin: 3rem 1.5rem;
+  `}
+
+  ${media.tablet`
+    margin: 5rem ${spacingLarge};
+  `}
+`;
+
+const theme = {
+  select: {
+    icons: {
+      color: colorBlue
+    }
+  },
+  global: {
+    colors: {
+      focus: colorBlue,
+      brand: colorBlue
+    },
+    input: {
+      weight: '400'
+    },
+    focus: {
+      border: {
+        color: colorBlue
+      }
+    }
+  }
+};
+
+const getAnswerTypes = (array, answerType) => array.find(item => item.type === answerType).values;
+
+const buildAnswerTypes = ({ answerType, name }, answerTypes) => {
+  const types = getAnswerTypes(answerTypes, answerType);
+  return types.map(option => {
+    return {
+      id: `${option}_${name}`,
+      value: option,
+      label: option
+    };
+  });
+};
+
+const renderFieldByType = (field, answerTypes, value, error, onChange, setFieldValue) => {
```

Comment:
you can always extract common part to avoid repetition and make it more easy to follow, like `args = {field, error, value}` and then apply it to `<Component {...args} />`

Reply:
[killing switch](https://glcheetham.name/2016/05/20/killing-switch-statements-in-react-with-the-strategy-pattern/) or [conditional rendering](https://www.robinwieruch.de/conditional-rendering-react/) not saying these are the best approachs, but let's kill the `switch`